docs/aws/audit/cloudtrailmonitoring/rules/cloudtrail_global_services_logging_duplicated_remediation.mdx

```diff
@@ -1,6 +1,297 @@
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access CloudTrail logs.
+
+2. Once AWS CLI is installed and configured, you can use the following command to list all the trails in your AWS account:
+
+   ```
+   aws cloudtrail describe-trails
+   ```
+   This command will return a JSON output with details of all the trails.
+
+3. To check for duplicate entries in CloudTrail logs, you can use the following command:
+
+   ```
+   aws cloudtrail lookup-events --lookup-attributes AttributeKey=EventName,AttributeValue=eventName
+   ```
+   Replace 'eventName' with the name of the event you want to check for duplicates. This command will return a list of all the events with the specified name.
+
+4. To detect duplicate entries, you can write a Python script that parses the JSON output from the above command and checks for duplicate entries. Here is a simple example:
+
+   ```python
+   import json
+   import subprocess
+
+   def get_cloudtrail_events(event_name):
+       command = ['aws', 'cloudtrail', 'lookup-events', '--lookup-attributes', 'AttributeKey=EventName,AttributeValue={}'.format(event_name)]
+       process = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+       stdout, stderr = process.communicate()
+       if process.returncode != 0:
+           raise Exception('Error executing command: {}'.format(stderr.decode('utf-8')))
+       return json.loads(stdout.decode('utf-8'))
+
+   def check_for_duplicates(events):
+       event_ids = [event['EventId'] for event in events['Events']]
+       if len(event_ids) != len(set(event_ids)):
+           print('Duplicate entries found')
+       else:
+           print('No duplicate entries found')
+
+   events = get_cloudtrail_events('eventName')
+   check_for_duplicates(events)
+   ```
+   Replace 'eventName' with the name of the event you want to check for duplicates. This script will print 'Duplicate entries found' if there are duplicate entries in the CloudTrail logs, and 'No duplicate entries found' otherwise.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. **Import necessary libraries and establish a session**: First, you need to import the necessary libraries in your Python script. You will need the boto3 library, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc. After importing the libraries, establish a session with AWS using your access key and secret access key.
+
+    ```python
+    import boto3
+
+    session = boto3.Session(
+        aws_access_key_id='YOUR_ACCESS_KEY',
+        aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>